### PR TITLE
patch bug in indicator that modified input datasets

### PR DESCRIPTION
**Pull request by nickeubank**
Previous version of indicator function would modify input datasets. This corrects that behavior and adds tests.

### CODE REVIEW

File: src/abstractdataframe/join.jl [hunk 1]
```diff
@@ -296,8 +296,8 @@ function Base.join(df1::AbstractDataFrame,
                  indicator_cols[i] *= 'X'
             end
          end
-        df1[Symbol(indicator_cols[1])] = trues(nrow(df1))
-        df2[Symbol(indicator_cols[2])] = trues(nrow(df2))
+         df1 = hcat(df1, DataFrame(Dict(indicator_cols[1] => [true for i in 1:nrow(df1)] )))
```
Comment (nalimilan) [inline src/abstractdataframe/join.jl]: Better keep using `trues`.

File: test/join.jl [hunk 2]
```diff
@@ -584,8 +584,13 @@ module TestJoin
                                     Name = ["John Doe", "Jane Doe", "Jane Doe", "Joe Blogs", missing],
                                     Job = ["Lawyer", "Doctor", "Florist", missing, "Farmer"],
                                     _merge = ["both", "both", "both", "left_only", "right_only"])
+
+        pre_join_name = copy(name)
```
Comment (nalimilan) [inline test/join.jl]: As noted in the other PR, why is this needed?

Comment (nickeubank) [reply test/join.jl]: Because `name` is being modified by `join`. that's the purpose of this patch. So to test and make sure that isn't happening in the future, we need a copy of `name` from before the join to compare to the post-join version of join.

File: src/abstractdataframe/join.jl (hunk 1 above)
Comment (nickeubank) [reply src/abstractdataframe/join.jl]: patched

File: test/join.jl (hunk 2 above)
Comment (nalimilan) [reply test/join.jl]: OK. These cases are difficult to decide: virtually all functions could modify their inputs, and we don't typically check this for all of them. Anyway if you prefer adding this please add a comment.

Comment (nickeubank) [reply test/join.jl]: My philosophy if that if a bug is found, when you patch it you add a test to ensure that the bug would be caught if it weren't fixed. Otherwise how do you know if your patch works? I'll add a comment.

**Comment by nickeubank** [comment]
thanks!
